kahuna/public/js/util/storage.js

```diff
@@ -0,0 +1,24 @@
+import angular from 'angular';
+
+export const storage  = angular.module('util.storage', []);
+
+storage.factory('storage', ['$window', function($window) {
+    function setItem(key, val) {
+        const storeVal = typeof val === 'string' ? val : JSON.stringify(val);
+        $window.localStorage.setItem(key, storeVal);
+    }
+
+    function getItem(key) {
+        let val = $window.localStorage.getItem(key);
+        try {
+            val = JSON.parse(val);
+        } catch (_) {}
```

Comment:
But if somehow the value got corrupted and you expected an Object...?

Do we need to store strings? Would make the interface simpler and more resilient only to allow Objects (incl Arrays)?